### PR TITLE
Pin byteorder on msrv

### PR DESCRIPTION
**Pull request by mejrs**
Fixes https://github.com/PyO3/pyo3/actions/runs/6435273809/job/17476207748?pr=3487

### CODE REVIEW

**Review by adamreichold — APPROVED** [review]

File: noxfile.py [hunk 1]
```diff
@@ -493,6 +493,7 @@ def set_minimal_package_versions(session: nox.Session):
         "regex": "1.7.3",
         "proptest": "1.0.0",
         "chrono": "0.4.25",
+        "byteorder": "1.4",
```
Comment (adamreichold) [inline noxfile.py:496]: I think you need a full version specification, e.g. `1.4.3`.

**Comment by davidhewitt** [comment]
I think to get this through CI you'll also need to resolve the new lint failure in https://github.com/PyO3/pyo3/actions/runs/6435273809/job/17476209992?pr=3487#step:7:1071

(caused by Rust 1.73 release)

**Comment by davidhewitt** [comment]
Ouch, @messense any idea what's going on in the macos failure in maturin-action?

As for why the UI test is only failing on windows coverage... 🤕

**Comment by messense** [comment]
> any idea what's going on in the macos failure in maturin-action?

Fixed in https://github.com/PyO3/maturin-action/pull/219

**Pull request merged**